beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -107,7 +107,9 @@ func (s *Store) insert(ctx context.Context,
 			s.headNode = n
 			s.highestReceivedNode = n
 		} else {
-			return n, errInvalidParentRoot
+			delete(s.nodeByRoot, root)
+			delete(s.nodeByPayload, payloadHash)
+			return nil, errInvalidParentRoot
```

Comment:
you could use `removeNode` here if you change how `removeNode` works by delete first then check if parent is nil, if the parent is nil then just return nil. I don't think it's worth the trouble though

Reply:
ah nice idea!